Makefile

```
@@ -96,8 +96,7 @@ style:
 hook:
 	# Install pre-commit git hook to check your changes for styleguide
 	# consistency.
-	cp kivy/tools/pep8checker/pre-commit.githook .git/hooks/pre-commit
-	chmod +x .git/hooks/pre-commit
+	pre-commit install -f
```

```suggestion
	pre-commit install -f --config=kivy/tools/pep8checker/.pre-commit-config.yaml
```

Can we please move the config file to `kivy/tools/pep8checker/.pre-commit-config.yaml` so that it's not polluting the root?

---

it's possible sure, but root of repo is the canonical location and the pre-commit framework has an optional post-clone -hook to check for that file and auto-install it.

---

Yeah, but I still would rather not pollute our project root with more config files than absolutely necessery. They accumulate quickly.

I guess we'll see what the others think.

---

Although root is the standard place,  there are pre-existing standards for kivy. I would 2nd @matham here. In this case,  kivy/tools/pep8checker is a cleaner and more consistent place.

---

Moved it to a separate directory as is does more than just pep8 and the old checker script should be removed after a deprecation period.